### ISSUE TITLE
allowdeprecated flag not effective or doesn't match messaging

### ISSUE THREAD

**Issue by buck54321** [zcash/zcash#6501]
Using version v5.4.2

I have a tmux session that runs multiple simnet nodes. I used to get addresses for the nodes to use in the script by running something like
```Bash
NODE_A_ADDR=`zcash-cli -conf=/path/to/node/a/conf getnewaddress`
```
`getnewaddress` is deprecated now, so I get an error message like 
```
Use z_getnewaccount and z_getaddressforaccount instead, or restart 
with `-allowdeprecated=getnewaddress` if you require backward compatibility.
```
Ideally, I could just use a unified address, but the return value from `z_getaddressforaccount` is JSON, and Debain does not have a native program for JSON parsing from the command-line, and I have many engineers who use these scripts. So to avoid complicating setup right now (kicking the can, I know), I'm trying to use the `-allowdeprecated=getnewaddress`. I've tried using the config option from both the command line on invocation of `zcashd`, and also by adding it to a configuration file. Both methods are ineffective, and I continue to receive the deprecation error.

**Comment by daira** [comment]
We will have a look at this. On the topic of command-line programs to parse JSON, I recommend `jq`. (The Debian package name is just `jq`.)

**Comment by buck54321** [comment]
Thanks @daira. I've worked around it for our testing harnesses, so no rush on my behalf. Just wanted to document the inconsistency.

**Comment by nuttycom** [comment]
I'm confused, because I think that this should have been remedied by https://github.com/zcash/zcash/pull/6423 which is included in v5.4.2. I've attempted to reproduce this locally but have been unable to do so; when running `./src/zcashd -allowdeprecated=getnewaddress` I'm able to use the `getnewaddress` as expected, and without the flag the RPC fails as expected.

**Comment by buck54321** [comment]
I can't reproduce anymore either. I must have had something confused. My bad.
